app/shop.py

```diff
@@ -0,0 +1,27 @@
+import datetime
+
+
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def calculate_product_cost(self, product_cart: dict) -> list:
```

Comment:
The return type hint for 'calculate_product_cost' is 'list', but it might be more descriptive to specify the type of elements in the list, for example, List[float] or List[int].